presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/thrift/ThriftHiveMetastore.java

```diff
@@ -1823,16 +1834,25 @@ public void updateTableWriteId(HiveIdentity identity, String dbName, String tabl
     }
 
     @Override
-    public void alterPartitions(HiveIdentity identity, String dbName, String tableName, List<Partition> partitions, long writeId)
+    public void alterPartitions(HiveIdentity identity, String dbName, String tableName, List<Partition> partitions, EnvironmentContext context, long writeId)
     {
         checkArgument(writeId > 0, "writeId should be a positive integer, but was %s", writeId);
         try {
             retry()
                     .stopOnIllegalExceptions()
                     .run("alterPartitions", stats.getAlterPartitions().wrap(() -> {
-                        try (ThriftMetastoreClient metastoreClient = createMetastoreClient(identity)) {
-                            metastoreClient.alterPartitions(dbName, tableName, partitions, writeId);
-                        }
+                        alternativeCall(
+                                () -> createMetastoreClient(identity),
```

Comment:
Do you need to close - terminate the metastoreclient before returning?
This may bring a connection - handle leak?

Reply:
This lambda function is being invoked inside the alternativeCall method from within a try-with-resources block which will take care of closing the connection.